includes/API/ApiLabkiUpdate.php

```diff
@@ -11,165 +11,336 @@
+            '_meta' => [ 'schemaVersion' => 1 ]
+        ];
     }
 
+    /* -------------------------------------------------------------------------
+     * UPDATE PACK
+     * ------------------------------------------------------------------------- */
+    private function handleUpdatePack(): array {
+        $params = $this->extractRequestParams();
+        $contentRepoUrl = (string)( $params['contentRepoUrl'] ?? '' );
+        $packName = (string)( $params['packName'] ?? '' );
+        $currentVersion = isset( $params['version'] ) ? (string)$params['version'] : null;
+        $newVersion = isset( $params['newVersion'] ) ? (string)$params['newVersion'] : null;
+        $pagesParam = $params['pages'] ?? null;
+        $pages = is_string( $pagesParam )
+            ? json_decode( $pagesParam, true )
+            : ( is_array( $pagesParam ) ? $pagesParam : [] );
+
+        if ( $contentRepoUrl === '' || $packName === '' ) {
+            $this->dieWithError( 'apierror-missing-param' );
+        }
+
+        $repoReg = new LabkiRepoRegistry();
+        $packReg = new LabkiPackRegistry();
+        $pageReg = new LabkiPageRegistry();
+
+        $repoId = $repoReg->ensureRepo( $contentRepoUrl );
+        $packIdObj = $packReg->getPackIdByName( $repoId, $packName, null );
+        if ( $packIdObj === null ) {
+            $this->dieWithError( [ 'apierror-pack-not-found', $packName ], 'pack_not_found' );
+        }
+        $packId = $packIdObj instanceof PackId ? $packIdObj->toInt() : (int)$packIdObj;
+
+        if ( $newVersion !== null && $newVersion !== '' ) {
+            $packReg->updatePack( $packId, [ 'version' => $newVersion ] );
+        } else {
+            $packReg->updatePack( $packId, [] );
+        }
+
+        // Replace pages
+        $pageReg->removePagesByPack( $packId );
+        $added = 0;
+        foreach ( $pages as $p ) {
+            if ( !is_array( $p ) ) {
+                continue;
+            }
+            $name = (string)( $p['name'] ?? '' );
+            $finalTitle = (string)( $p['finalTitle'] ?? '' );
+            $ns = (int)( $p['pageNamespace'] ?? 0 );
+            $wikiPageId = (int)( $p['wikiPageId'] ?? 0 );
+            if ( $name === '' || $finalTitle === '' ) {
+                continue;
+            }
+            $pageReg->recordInstalledPage( $packId, $name, $finalTitle, $ns, $wikiPageId );
+            $added++;
+        }
+
+        return [
+            'success' => true,
+            'action' => 'updatePack',
+            'packId' => $packId,
+            'pagesAdded' => $added,
+            '_meta' => [ 'schemaVersion' => 1 ]
+        ];
+    }
+
+    /* -------------------------------------------------------------------------
+     * REMOVE PACK
+     * ------------------------------------------------------------------------- */
+    private function handleRemovePack(): array {
+        $params = $this->extractRequestParams();
+        $packId = (int)( $params['packId'] ?? 0 );
```

Comment:
typically it's a good idea to give the api caller one and only one way to refer to something: a pack is defined externally as a (repo_uri, pack_name) tuple, and the packID is an internal implementation detail. seems easier to me to just always parse the (repo_uri, pack_name) than to juggle both. or if we want to do that, wrap that in a method so it's only written once and applies to all the variants of this api endpoint